src/components/AttachmentModal.tsx

```diff
@@ -1,186 +1,218 @@
+};
 
-    /** Optional original filename when uploading */
-    originalFileName: PropTypes.string,
+type ImagePickerResponse = {
+    height: number;
+    name: string;
+    size: number;
+    type: string;
+    uri: string;
+    width: number;
+};
 
-    /** A function as a child to pass modal launching methods to */
-    children: PropTypes.func,
+type Data = File | ImagePickerResponse;
+
+type ChildrenProps = {
+    displayFileInModal: (data: Data) => void;
+    show: () => void;
+};
+
+type AttachmentModalProps = AttachmentModalOnyxProps & {
+    /** Optional source (URL, SVG function) for the image shown. If not passed in via props must be specified when modal is opened. */
+    source?: string;
+
+    /** Optional callback to fire when we want to preview an image and approve it for use. */
+    onConfirm?: ((file: Partial<File>) => void) | null;
+
+    /** Whether the modal should be open by default */
+    defaultOpen?: boolean;
+
+    /** Optional original filename when uploading */
+    originalFileName?: string;
 
     /** Whether source url requires authentication */
-    isAuthTokenRequired: PropTypes.bool,
+    isAuthTokenRequired?: boolean;
 
     /** Determines if download Button should be shown or not */
-    allowDownload: PropTypes.bool,
+    allowDownload?: boolean;
 
     /** Title shown in the header of the modal */
-    headerTitle: PropTypes.string,
+    headerTitle?: string;
 
     /** The report that has this attachment */
-    report: reportPropTypes,
+    report?: OnyxTypes.Report;
 
-    /** The transaction associated with the receipt attachment, if any */
-    transaction: transactionPropTypes,
+    /** Optional callback to fire when we want to do something after modal show. */
+    onModalShow?: () => void;
 
-    ...withLocalizePropTypes,
+    /** Optional callback to fire when we want to do something after modal hide. */
+    onModalHide?: () => void;
 
-    ...windowDimensionsPropTypes,
+    /** Optional callback to fire when we want to do something after attachment carousel changes. */
+    onCarouselAttachmentChange?: (attachment: Attachment) => void;
 
     /** Denotes whether it is a workspace avatar or not */
-    isWorkspaceAvatar: PropTypes.bool,
+    isWorkspaceAvatar?: boolean;
 
     /** Whether it is a receipt attachment or not */
-    isReceiptAttachment: PropTypes.bool,
-};
+    isReceiptAttachment?: boolean;
 
-const defaultProps = {
-    source: '',
-    onConfirm: null,
-    defaultOpen: false,
-    originalFileName: '',
-    children: null,
-    isAuthTokenRequired: false,
-    allowDownload: false,
-    headerTitle: null,
-    report: {},
-    transaction: {},
-    onModalShow: () => {},
-    onModalHide: () => {},
-    onCarouselAttachmentChange: () => {},
-    isWorkspaceAvatar: false,
-    isReceiptAttachment: false,
+    /** A function as a child to pass modal launching methods to */
+    children?: React.FC<ChildrenProps>;
+
+    fallbackSource?: AvatarSource | number;
```

Comment:
AvatarSource type includes `number`
```suggestion
    fallbackSource?: AvatarSource;
```